src/visitable.cpp

```diff
@@ -774,7 +782,8 @@ static int amount_of_internal( const T& self, const itype_id& id, bool pseudo, i
 {
     int qty = 0;
     self.visit_items( [&qty, &id, &pseudo, &limit] ( const item *e ) {
-        qty += ( e->typeId() == id && e->allow_crafting_component() && ( pseudo || !e->has_flag( "PSEUDO" ) ) );
+        bool is_component = e->typeId() == id && e->allow_crafting_component() && ( pseudo || !e->has_flag( "PSEUDO" ) );
+        qty = accumulate_with_limit( qty, is_component, limit );
```

Comment:
Operators aren't going to make this any better, your options for turning a bool into a conditional increment are either use the boolean as-is and have it implicitly added after a cast, or hide the increment behind a branch.  IMO implicitly casting to int is far cleaner.